### ISSUE TITLE
We stopped receiving communications from the remote host.

### ISSUE THREAD

**Issue by high-jin** [ValveSoftware/Dota2-Gameplay#21838]
### Description

私は日本人です。
英文は全てDeepLでの翻訳文です。
変な言葉遣いになっていたらごめんなさい。
現在私の環境では、各ヒーローのデモプレイや、CPUとのゲームは正常に動作していますが、カスタムロビーやノーマルマッチ、TURBO、ランクマッチでのゲームプレイができない状態に陥っています。
他プレイヤーとのマッチングは成功しますが、ヒーローピック画面に遷移した後、ヒーローの一覧が表示されないまま、一定時間後にゲームから退出してしまいます。
また、ヒーローの一覧は表示されませんが、画面下部のチャットウィンドウは操作できます。
ゲーム開始ができず、失敗するたびに一定時間のペナルティを課せられてしまいます。
少なくとも2023/8/23にゲーム自体は行えていることを確認しています。
私の環境に不備があるのか、何かしらのブロックを受けているのか、何もわからず、可能であればゲームプレイできるよう問題解消をしたく考えています。
同様のエラーが起きて復旧できた方、明確な復旧方法がわかっているなど、ありませんでしょうか？
添付の画像は、カスタムロビーでのエラーパターンのものになります。

******************

I am Japanese.
All English text is translated from the original text at DeepL.
Sorry if the wording is strange.
Currently in my environment, the demo play of each hero and games with the CPU are working fine, but I am stuck in a state where I cannot play in the custom lobby, normal matches, TURBO, or ranked matches.
Matching with other players is successful, but after transitioning to the hero pick screen, the player exits the game after a certain amount of time without a list of heroes being displayed.
Also, the list of heroes is not displayed, but the chat window at the bottom of the screen can be operated.
The game cannot be started and each failed attempt is penalized for a certain amount of time.
I have confirmed that the game itself can be played at least on 8/23/2023.
I am not sure if there is something wrong with my environment or if I am being blocked in some way or what, and would like to resolve the problem so that I can play the game if possible.
Is there anyone who has had a similar error and was able to recover from it, or knows a clear way to recover?
The attached image is of the error pattern in the custom lobby.

### Screenshots

![error01](https://github.com/user-attachments/assets/1ae1baa6-d756-41c8-a2b3-e9bd143766e6)
![error02](https://github.com/user-attachments/assets/526d5a96-283c-4efe-9b13-6440357865b7)
![error03](https://github.com/user-attachments/assets/f5b4631e-88b1-488a-86fb-53615f535d8a)
[error04.txt](https://github.com/user-attachments/files/17024624/error04.txt)
[console.log](https://github.com/user-attachments/files/17024627/console.log)

**Comment by commentator24** [comment]
#21547 someone hv same problem and said reinstall / format disk didnt help.

I am not dev and i dont know much about console log errors, but i looked at your error txt file.
To me it looks like errors related to bots skill/items/scripts not loading. no valid headings. wrong index? resources not loading
but the biggest issue i think your vport stays 0 so maybe firewall/modem not configured correctly. router issue?

__vport 0?__
chatgpt says Thread starvation is a situation in multithreading environments where a thread is perpetually denied the resources it needs to execute, typically because other threads are continuously being prioritized over it. This can lead to performance issues and delays in program execution. Here’s a more detailed breakdown:

Causes of Thread Starvation
Priority Scheduling: If the scheduling algorithm favors higher-priority threads, lower-priority threads may never get CPU time, leading to starvation.
Resource Contention: When multiple threads are competing for limited resources (like CPU, memory, or I/O), some threads may be unable to acquire the necessary resources to proceed.
Improper Thread Management: Poorly designed algorithms or improper use of synchronization primitives (like mutexes or locks) can lead to scenarios where certain threads are perpetually blocked.
Infinite Loops or Long-Running Tasks: If a thread is stuck in an infinite loop or a long-running task, it may prevent other threads from running if they require the same resources.

[SteamNetSockets] Received Steam datagram ticket for server steamid:90201819581986816 vport 0. match_id=7946152773 mm=0 flg=
[SteamNetSockets] SteamNetworkingSockets lock held for 11.8ms.  (Performance warning.)  
This is usually a symptom of a general performance problem such as thread starvation.
[GCClient] Recv msg 26 (k_ESOMsg_UpdateMultiple), 1734 bytes
[Console] Saving lobby data to disk.
WriteSteamRemoteStorageFileAsync( "scripts/dota_local_disk_lobby_data.txt" ) -> at 78.256
Telling Steam not to update the app for the next 900 seconds
Job EconItemSearch has spent >3.000ms without yielding: 10.004ms
[SteamNetSockets] [#114674209 SDR server steamid:90201819581986816 vport 0 'server'] 7 reply timeouts, last recv 6256.7ms ago.  Sending keepalive.
[SteamNetSockets] [#114674209 SDR server steamid:90201819581986816 vport 0 'server'] 16 reply timeouts, last recv 11679.5ms ago.  Sending keepalive.
[SteamNetSockets] [#114674209 SDR server steamid:90201819581986816 vport 0 'server'] Timed out.  11851.7ms since last recv, 10013.6ms since timeout started, 16 consecutive failures

__resources not loading properly? also interesting your panel language trying to localise as Indonesian?__
[AnimResource]  models/items/skywrath_mage/skywrath_arcana/skywrath_arcana_pedestal.vmdl : Animations loaded but no animation data found
Excessive frame time of 143.46ms clamped.  Acting as if the frame only took 100.00ms
Hero npc_dota_hero_tiny is using an invalid ability special_bonus_strength_10 in its bot skill build!
[ResourceSystem] Failed loading resource "panorama/images/spellicons/dota_base_ability_png.vtex_c" (ERROR_FILEOPEN: File not found)
[ResourceSystem] Failed loading resource "scripts/ping_wheel_staging.vdata_c" (ERROR_FILEOPEN: File not found)
Novel Dialogue snake-charmer-tokens-earnable has no valid heading.
ILocalize::AddFile() failed to load file "resource/localization/broadcastfacts_english.txt".
ILocalize::AddFile() failed to load file "resource/localization/broadcastfacts_japanese.txt".
**** Unable to localize '#GameUI_Language_Indonesian' on panel 'indonesian'
[AnimResource]  models/items/skywrath_mage/skywrath_arcana/skywrath_arcana_pedestal.vmdl : Animations loaded but no animation data found
[Client] Client missing networkable entity class CLightGlow
[Client] Client missing networkable entity class CMoverPathNode
[Client] Client missing networkable entity class CPathMover
[Client] Client missing networkable entity class CPhysMagnet
[Client] Client missing networkable entity class CSpotlightEnd
[Client] LoadingDisplay changed from NONE to LOADING (map="")

[SteamNetSockets] SDR network config fetch attempt #1 failed.  HTTP 504.  .  Trying again.
[Localization System] CLocalize::FindSafe failed to localize: invalid index
**** Panel TimeRangeSlider has fill-parent-flow for width, but isn't in a flowing right layout
**** Panel TimeScrubSlider has fill-parent-flow for width, but isn't in a flowing right layout
**** Panel SliderAndMarkers has fill-parent-flow for width, but isn't in a flowing right layout
**** Panel TimeScrubSlider has fill-parent-flow for width, but isn't in a flowing right layout

__cannot receive packet from server?__
[Server] SV:  Sending full update to client High-jin (reason:  initial update)
[Client] CNetworkGameClientBase::OnReceivedUncompressedPacket(), received full update
[Client] Receiving 0.0K non-incremental update from server

[Networking] Timeout; remote problem. Rx age server 11.9s relay 0.2s
[NetSteamConn] Closing Steam Net Connection on socket 'client' to =[A:1:3058120704:30227], handle #114674209 (0 (null))
[NetSteamConn] CloseSteamNetConnection handle #114674209 (userdata 0)
[NetSteamConn] Disassociating NetChan server @ =[A:1:3058120704:30227] (=[A:1:3058120704:30227]) from Steam Net Connection handle #114674209 (userdata 0)
[NetSteamConn] Removing Steam Net Connection for =[A:1:3058120704:30227], handle #114674209
[Client] Disconnection during connection phase. Sign-on state: 5 (SIGNONSTATE_SPAWN). Disconnect reason: 79 (NETWORK_DISCONNECT_REMOTE_TIMEOUT).

Try go to your steam setting and change download region and downloads cache. if you're from japan you have few options like tokyo or osaka.
![image](https://github.com/user-attachments/assets/0e29a7df-4dde-4ba0-9af3-96f858113c7f)
![image](https://github.com/user-attachments/assets/8d2d8554-29c6-4218-bf39-1910dfb96f5f)

for language you can set on dota and steam. i would recommend you download both english and japanese in case there are any errors? e.g. select english first, close, Verify integrity of dota 2 files then change to japanese. The language for dota will affect voiceovers. steam will affect text i think.

![image](https://github.com/user-attachments/assets/875a72b3-b7c4-467f-b3f0-cf70eb1d24e8)
![image](https://github.com/user-attachments/assets/60533180-f80a-455c-ab81-f2bada3a6d8a)
since you had wrong index, maybe missing file/memory issue/cannot read memory?
Try verify integrity.. disk cleanup and defrag..also make sure your free space is 4gb and 4gb virtual memory.
then go to your modem setting e.g. typing 192.168.1.1 or whatever your ISP use, and add quality of service for dota 2 to fix router issue..also check windows firewall and your windows user account has admin rights for permission etc
if you found out something let me know so i learn too aha

**Comment by kylonie** [comment]
Having the same issue since the last update. I've tried so many ways to solve it but none of them worked. Idk what to do rn and I haven't played dota for so long because of this issue.

**Comment by commentator24** [comment]
@kylonie did you try what i said? clean your downloads cache and change region
and is your steam language set to english or other languages?

**Comment by kylonie** [comment]
@commentator24 yep, I tried them and none of them worked too and I'm only using English for steam language.

**Comment by commentator24** [comment]
then have you added dota 2 app on QoS (quality of service) in your modem setting?

**Comment by Izeath** [comment]
Did the exact same thing. Nothing works. Able to watch other games inside Dota2 but unable to play any match besides Play with Bots.

**Comment by kylonie** [comment]
@commentator24 I guess that there's nothing wrong with the modem or connection/network, 'cause my brother could able to play dota that is connected on the same modem and it works fine.

**Comment by Izeath** [comment]
I created a new account to check. Not based on the account. before they release the facets I was able to play smoothly. after that it went downhill. During the Patch where facets was released, no hero selection and no error message. But recent patch it gives this network error. Still same problem, cant play a Dota 2.

**Comment by kylonie** [comment]
> I created a new account to check. Not based on the account. before they release the facets I was able to play smoothly. after that it went downhill. During the Patch where facets was released, no hero selection and no error message. But recent patch it gives this network error. Still same problem, cant play a Dota 2.

Yeah, that's what I was thinking too, 'cause even before the patch (when they released Ringmaster) I can play pub matches and I can even play ranked too but when I came back playing after the patch (releasing of Ringmaster) this server error message keeps on popping up for whatever the reason is and even when I queue for new player mode it still goves me the same message. Now I'm still figuring out what causes the error and I tried a lot of advices and tips and none of them worked.

**Comment by high-jin** [comment]
皆さんありがとうございます。
やはりなにをしても症状は変わりませんでした。
モバイルルーターやiPhoneのテザリング機能で経由しても変わりません。
パソコンに入っている特定のアプリケーションでブロックされているか、アカウント単位でブロックされているかしか思いつきません。
何度かDota2自体を再インストールし直しても、ログの内容に差異が生まれないので、インストール内容が間違っていない限りはその様な理由だと思います。
新しいパソコンを用意できたら再チャレンジしますが、それまでは改善策が見つかるまで待つしかないかもしれません。

********************

Thank you all.
As I thought, no matter what I did, the symptoms did not change.
It does not change even if I use a mobile router or the tethering function on my iPhone.
The only thing I can think of is that it is being blocked by a specific application on my computer or by account.
Re-installing Dota2 several times does not make any difference in the logs, so unless I've installed something wrong, I think that's the reason.
I will try again when I can get a new computer, but until then I may have to wait until I find a remedy.

**Comment by kylonie** [comment]
> 皆さんありがとうございます。 やはりなにをしても症状は変わりませんでした。 モバイルルーターやiPhoneのテザリング機能で経由しても変わりません。 パソコンに入っている特定のアプリケーションでブロックされているか、アカウント単位でブロックされているかしか思いつきません。 何度かDota2自体を再インストールし直しても、ログの内容に差異が生まれないので、インストール内容が間違っていない限りはその様な理由だと思います。 新しいパソコンを用意できたら再チャレンジしますが、それまでは改善策が見つかるまで待つしかないかもしれません。
> 
> Thank you all. As I thought, no matter what I did, the symptoms did not change. It does not change even if I use a mobile router or the tethering function on my iPhone. The only thing I can think of is that it is being blocked by a specific application on my computer or by account. Re-installing Dota2 several times does not make any difference in the logs, so unless I've installed something wrong, I think that's the reason. I will try again when I can get a new computer, but until then I may have to wait until I find a remedy.

Yeah, I hope Valve will do something about this issue. I'm still trying to figure out what could be the problem but it seems that there's a lot of us experiencing it. I'm still hoping that someone can help us get through this issue.

**Comment by Izeath** [comment]
Could it be related to our Processor? or GPU? Still I'm able to play other games aside Dota 2. My rig is old but still works well with other games.

**Comment by commentator24** [comment]
no i dont think it's processor or GPU. it's more likely network memory/cache issue. e.g. after an update the cache is not updating / cleared. Reinstalling would not help because cache is still saved somewhere. 

chatgpt says If reinstalling Dota 2 didn’t help, there could be cached data, network configurations, or leftover files affecting your connection. Clearing the Steam cache, DNS cache, and restarting your router can help resolve these issues. The Steam client has its own cache that might retain outdated information even after you reinstall the game.

try disable your firewall/antivirus or whitelist it, add dota 2 app to exception.
Then flush DNS: Open Command Prompt as an administrator and type ipconfig /flushdns, then press Enter
and is your ip address static or dynamic? look for DHCP setting in your router/modem setting. 
you can also simply open command prompt and type ipconfig /all to see. if DHCP enabled Yes then your IP address is dynamic.
Static IP might help, but usually dynamic is also fine.

Flushing DNS cache removes all entries stored in your device's DNS resolver cache. This cache keeps records of previously resolved domain names and their corresponding IP addresses to speed up future requests.

When you flush the DNS cache, it helps resolve issues like:
Outdated Records: If a website has changed its IP address, flushing the cache ensures your device queries the DNS server for the most current address.
Connection Problems: It can resolve issues related to connectivity or loading errors that may arise from corrupt or outdated DNS information. 
Changes in DNS Settings: If you've recently changed DNS settings (like switching to a different DNS provider), flushing the cache helps your device recognize those changes immediately.

I would also try disk cleanup, defrag, clean steam downloads cache, and flushdns maybe theres some old data messing around.

**Comment by Izeath** [comment]
Yeah, I did exactly what you suggested. Disk cleanup, defrag, clean steam cache, flushdns and even removed any files relating to dota 2. By this time dota 2 is un-installed. Re-installed the game but same result. Bad patch I guess, since lots of players are experiencing this. *sigh*

**Comment by commentator24** [comment]
did you try static IP address?

**Comment by Izeath** [comment]
Yep, changed to static still have the same problem.

**Comment by kylonie** [comment]
@commentator24 tried it all too, the result is still the same.

**Comment by schawarzer** [comment]
tried to email gabben, hoping to fix this or give us a solution.

**Comment by kylonie** [comment]
Are there any updates guys?

**Comment by Izeath** [comment]
Man, even after the recent patch still having issues. Any good news?

**Comment by schawarzer** [comment]
UPDATE:  I've tried uninstalling steam but still the same result

**Comment by commentator24** [comment]
i think reinstall steam or reinstall dota wont help because cache files are not deleted and your settings saved

maybe you can call your ISP and ask them to ping to the server is there any issue from their end?
chatgpt also say maybe your Network drivers are outdated or has settings like idle timeout / energy efficient ethernet enabled by default. e.g. some device has Low power state and dynamic power adjustment / TCP timeout / session timeout settings in router configuration interface that leads to faster disconnection of idle connections / reduce power consumption with low-power mode. the device might limit its activity to conserve energy and cause delay in re-establishing the connection, especially if connection was idle, leading to perceived disconnections or dropped connections.

Network Interface Card (NIC) Settings:
check the Advanced tab for any settings related to power management or idle timeouts.
go to your Device manager > network adapters > right click the ethernet/LAN for wire or wifi for wireless. rightclick properties
![image](https://github.com/user-attachments/assets/c8d13de5-111b-4ccf-af70-1239b355e882)

TCP keep alive settings
Open regedit and navigate to HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters.
Look for KeepAliveTime and KeepAliveInterval. You can adjust these values to change how often keep-alive packets are sent.

Router/Switch Timeout Settings: access your router e.g. enter IP address in web browser
Look for sections related to TCP Timeout, Session Timeout. These settings define how long an idle connection is maintained before being dropped.
also check Quality of Service (QoS) Settings: this prioritize traffic and dictate how idle connections are handled. 

Firewall Settings:
also check your firewall setting options that specify idle connection timeouts. Adjust these to control how long idle connections are kept alive.

**Comment by kylonie** [comment]
> i think reinstall steam or reinstall dota wont help because cache files are not deleted and your settings saved
> 
> maybe you can call your ISP and ask them to ping to the server is there any issue from their end? chatgpt also say maybe your Network drivers are outdated or has settings like idle timeout / energy efficient ethernet enabled by default. e.g. some device has Low power state and dynamic power adjustment / TCP timeout / session timeout settings in router configuration interface that leads to faster disconnection of idle connections / reduce power consumption with low-power mode. the device might limit its activity to conserve energy and cause delay in re-establishing the connection, especially if connection was idle, leading to perceived disconnections or dropped connections.
> 
> Network Interface Card (NIC) Settings: check the Advanced tab for any settings related to power management or idle timeouts. go to your Device manager > network adapters > right click the ethernet/LAN for wire or wifi for wireless. rightclick properties Dont forget to right click update driver as well. you can roll back if needed ![image](https://private-user-images.githubusercontent.com/109324155/375116282-c8d13de5-111b-4ccf-af70-1239b355e882.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NjYzMDYsIm5iZiI6MTcyODU2NjAwNiwicGF0aCI6Ii8xMDkzMjQxNTUvMzc1MTE2MjgyLWM4ZDEzZGU1LTExMWItNGNjZi1hZjcwLTEyMzliMzU1ZTg4Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxMFQxMzEzMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYWJlZDlmMWUxOGQzODljMjI5MzNiMDUwNzgyMTQ0ZTgzZWYyZjRmZWNmZmRhNzg3MTU1M2M3NDk0ZWMzMTkyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.iG777EoYAX8pmipJILCYpGApUUJRsz8yvDXWwV9eKYI)
> 
> TCP keep alive settings Open regedit and navigate to HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters. Look for KeepAliveTime and KeepAliveInterval. You can adjust these values to change how often keep-alive packets are sent.
> 
> Router/Switch Timeout Settings: access your router e.g. enter IP address in web browser Look for sections related to TCP Timeout, Session Timeout. These settings define how long an idle connection is maintained before being dropped. also check Quality of Service (QoS) Settings: this prioritize traffic and dictate how idle connections are handled.
> 
> Firewall Settings: also check your firewall setting options that specify idle connection timeouts. Adjust these to control how long idle connections are kept alive.

@commentator24 I don't think there is an issue with our network or connection, 'cause how is my brother be able to play Dota even though we are connected on the same network or connection? We're both connected to our router through ethernet. That's why I'm getting a little confused rn since why can my brother be able to play while mine is getting blocked or disrupted by the server?

**Comment by schawarzer** [comment]
well the only clue i have is out of many accounts i borrow to test which is working only 1 account works out of 5 accounts ive tried and that was my brothers account while my account cant even play on the same device.

**Comment by schawarzer** [comment]
Good news i have new clue!!

i can finally join a match.

what i did: create new USER on my PC then run steam>dota2 in that user 


i hope someone can narrow down the problem cause.

**Comment by high-jin** [comment]
Are you saying that the problem is tied to user information?
This is a problem that is out of the average person's control.
I've tried to contact support, but they just dismiss it as a communication problem, and tell me to throw away all my skins and game results and play with a new account.

**Comment by high-jin** [comment]
I'll open it because I closed it by mistake.

**Comment by commentator24** [comment]
Maybe related to Steam cloud data? disable this temporarily to see if it resolves the issue.
or maybe windows user account do not have admin permission?

**Comment by Izeath** [comment]
> Good news i have new clue!!
> 
> i can finally join a match.
> 
> what i did: create new USER on my PC then run steam>dota2 in that user
> 
> i hope someone can narrow down the problem cause.

Nah didn't work for me. I even disabled the Steam Cloud.

**Comment by kylonie** [comment]
> Good news i have new clue!!
> 
> i can finally join a match.
> 
> what i did: create new USER on my PC then run steam>dota2 in that user
> 
> i hope someone can narrow down the problem cause.

Didn't work for me either.

**Comment by Izeath** [comment]
I have a question, I tried reviewing the console while joining a match.
How do you fix or find this file?

[ResourceSystem] Failed loading resource "panorama/images/players/0_png.vtex_c" (ERROR_FILEOPEN: File not found)

Thank you!

**Comment by kylonie** [comment]
> I have a question, I tried reviewing the console while joining a match. How do you fix or find this file?
> 
> [ResourceSystem] Failed loading resource "panorama/images/players/0_png.vtex_c" (ERROR_FILEOPEN: File not found)
> 
> Thank you!

I got that error too when I was observing my console log.

**Comment by Izeath** [comment]
Any luck? Haven't tried playing dota 2 for a while now.

**Comment by commentator24** [comment]
try configure DNS to something reliable like google DNS maybe? what is it currently on in your modem setting

**Comment by Izeath** [comment]
I already did this. Changed it to Cisco DNS, Google DNS and others still the same issue. Thing is, it was working in the past. Right after the patch that added the facets etc. this keeps on happening.

**Comment by fasjkfhakjsfh** [comment]
I disabled "AMD Radeon Anti-Lag 2.0" under video options, and it fixed this. I was having trouble skipping the loading animation on start-up too though.

**Comment by Izeath** [comment]
I tried restoring my 3D Settings in my NVIDIA CONTROL PANEL. I'm finally playing Dota 2 again. Dunno if the recent update helped. How bout you guys?

**Comment by Nirukooo** [comment]
> I disabled "AMD Radeon Anti-Lag 2.0" under video options, and it fixed this. I was having trouble skipping the loading animation on start-up too though.

Thank you very muuuuch it fixed !!!!!!

**Comment by pwhug** [comment]
I fixed this issue on my pc.

I switched to Vulkan (dlc) instead of Direct X-whatever.

Right click dota in steam library > properties > DLC > check the box next to Dota 2 Vulkan Support. It will download and install.

Then start dota > video settings > Options > use the dropdown menu to select Vulkan and restart dota.

This fixed my match loading bug and also improved performance in the game.

**Comment by AnanasSok** [comment]
Changing the rendering api to vulkan helped me. If you want to stay on dx11, then resetting the shader cache may help (it helped me). AMD software has a separate function for this, for nvidia you need to look for guides on the Internet
